indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -91,151 +97,330 @@ public WorkerTaskMonitor(
   public void start()
   {
     try {
-      // restore restorable tasks
-      final List<Pair<Task, ListenableFuture<TaskStatus>>> restored = taskRunner.restore();
-      for (Pair<Task, ListenableFuture<TaskStatus>> pair : restored) {
-        submitTaskRunnable(pair.lhs, pair.rhs);
-      }
-
-      // cleanup any old running task announcements which are invalid after restart
-      for (TaskAnnouncement announcement : workerCuratorCoordinator.getAnnouncements()) {
-        if (!isTaskRunning(announcement.getTaskStatus().getId()) && announcement.getTaskStatus().isRunnable()) {
-          workerCuratorCoordinator.updateAnnouncement(
-              TaskAnnouncement.create(
-                  announcement.getTaskStatus().getId(),
-                  announcement.getTaskResource(),
-                  TaskStatus.failure(announcement.getTaskStatus().getId())
-              )
-          );
-        }
-      }
-
-      pathChildrenCache.getListenable().addListener(
-          new PathChildrenCacheListener()
+      restoreRestorableTasks();
```

Comment:
I don't think start/stop can ever be called simultaneously with how Druid normally starts things. but, sure.